### PR TITLE
Added separate dispatcher

### PR DESCRIPTION
**Pull request by 1e-to**

### CODE REVIEW

File: numba/dppl/gpu_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from numba.dppl.compiler import DPPLCompiler
+from numba.core.cpu_options import ParallelOptions
+
+
+class GPUDispatcher(dispatcher.Dispatcher):
+    targetdescr = cpu_target
+
+    def __init__(self, py_func, locals={}, targetoptions={}, impl_kind='direct', pipeline_class=compiler.Compiler):
+        if dppl_config.dppl_present:
+            targetoptions['parallel'] = {'offload':True}
```
Comment (AlexanderKalistratov) [inline numba/dppl/gpu_dispatcher.py]: why do we need this option?

**Comment by AlexanderKalistratov** [comment]
Let's also cleanup cpu options:
https://github.com/IntelPython/numba/blob/pydppl/numba/core/cpu_options.py#L48

**Comment by 1e-to** [comment]
@diptorupd , @reazulhoque  could you please take a look?

File: numba/core/decorators.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from numba.core.errors import DeprecationError, NumbaDeprecationWarning
 from numba.stencils.stencil import stencil
 from numba.core import config, sigutils, registry, cpu_dispatcher
+from numba.dppl import dppl_offload_dispatcher
```
Comment (AlexanderKalistratov) [inline numba/core/decorators.py]: Let's remove this line. It is not needed

Comment (1e-to) [reply numba/core/decorators.py]: It is needed for our target to be added to the registry

**Comment by diptorupd** [comment]
@AlexanderKalistratov @1e-to Is this ready to go? 

Also, I will be opening a PR to change Numba to use the new `dpctl` instead of the `dpctl.ocldrv`. I will rebase my changes on top of this PR.

**Comment by AlexanderKalistratov** [comment]
@diptorupd it is ready

File: numba/core/cpu_dispatcher.py [hunk 3]
```diff
@@ -6,22 +6,8 @@ class CPUDispatcher(dispatcher.Dispatcher):
...
-                print("---------------------------------------------------------------------------")
-                print("WARNING : offload=True option ignored. Ensure OpenCL drivers are installed.")
-                print("---------------------------------------------------------------------------")
-                dispatcher.Dispatcher.__init__(self, py_func, locals=locals,
-                    targetoptions=targetoptions, impl_kind=impl_kind, pipeline_class=pipeline_class)
-        else:
-            dispatcher.Dispatcher.__init__(self, py_func, locals=locals,
-                targetoptions=targetoptions, impl_kind=impl_kind, pipeline_class=pipeline_class)
+        dispatcher.Dispatcher.__init__(self, py_func, locals=locals,
+            targetoptions=targetoptions, impl_kind=impl_kind, pipeline_class=pipeline_class)
```
Comment (reazulhoque) [inline numba/core/cpu_dispatcher.py:10]: This piece of code was refactored from numba/core/registry.py. May be we should put it back there.

File: numba/dppl/dppl_offload_dispatcher.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from numba.core import dispatcher, compiler
+from numba.core.registry import cpu_target, dispatcher_registry
+import numba.dppl_config as dppl_config
+from numba.dppl.compiler import DPPLCompiler
+from numba.core.cpu_options import ParallelOptions
+
+
+class DpplOffloadDispatcher(dispatcher.Dispatcher):
+    targetdescr = cpu_target
```
Comment (reazulhoque) [inline numba/dppl/dppl_offload_dispatcher.py:7]: I think we should create our own target that inherits from the CPUTarget. We already have a dppl target but that target is specifically for the GPU code generation. This target will be the target that takes care of any CPU code we might need (like wrappers and calls made to the GPU code)

**Comment by 1e-to** [comment]
@reazulhoque, I totally agree with the changes you are proposing. Let's merge the first working version now, and after the beta release, I will immediately correct the code in accordance with your comments?

**Pull request merged**